src/VisualStudio/Core/Def/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs

```diff
@@ -322,19 +296,8 @@ public TableEntriesSource(LiveTableDataSource source, Workspace workspace, IGlob
 
             public override ImmutableArray<DiagnosticTableItem> GetItems()
             {
-                // If we're in lsp pull mode then lsp client owns the error list.
-                var diagnosticMode = _globalOptions.GetDiagnosticMode();
-                if (diagnosticMode == DiagnosticMode.LspPull)
-                    return ImmutableArray<DiagnosticTableItem>.Empty;
-
-                var provider = _source._diagnosticService;
-                var items = provider.GetDiagnosticsAsync(_workspace, _projectId, _documentId, _id, includeSuppressedDiagnostics: true, CancellationToken.None)
-                    .AsTask()
-                    .WaitAndGetResult_CanCallOnBackground(CancellationToken.None)
-                                    .Where(ShouldInclude)
-                                    .Select(data => DiagnosticTableItem.Create(_workspace, data));
-
-                return items.ToImmutableArray();
+                // Since we're in lsp pull mode the lsp client owns the error list.
```

Comment:
We support adding source suppressions (pragma/SuppressMessageAttribute) for Roslyn diagnostics from the error list, see documentation [here](https://learn.microsoft.com/visualstudio/code-quality/in-source-suppression-overview?view=vs-2022&tabs=csharp#suppress-violations-in-source-code) in `From the Error List` section. I believe this feature does not rely on whether the error list entries come from Roslyn or VS Platform, but we may want to double check that this feature is not broken in LSP pull diagnostics mode.

Reply:
gotcha.  yeah, that may be something we have to decide what we want to do with in the future if it doesn't work.